File: seed/ruby-sdk-v2/pagination/lib/seed/inline_users/inline_users/types/user_optional_list_page.rb
# frozen_string_literal: true

module Seed
  module InlineUsers
    module InlineUsers
      module Types
        class UserOptionalListPage < Internal::Types::Model
          field :data, lambda {
            Seed::InlineUsers::InlineUsers::Types::UserOptionalListContainer
          }, optional: false, nullable: false
          field :next_, -> { String }, optional: true, nullable: false, api_name: "next"
        end
      end
    end
  end
end

File: seed/php-model/pagination/src/InlineUsers/InlineUsers/Traits/UserOptionalListPage.php
<?php

namespace Seed\InlineUsers\InlineUsers\Traits;

use Seed\InlineUsers\InlineUsers\UserOptionalListContainer;
use Seed\Core\Json\JsonProperty;

/**
 * @property UserOptionalListContainer $data
 * @property ?string $next
 */
trait UserOptionalListPage 
{
    /**
     * @var UserOptionalListContainer $data
     */
    #[JsonProperty('data')]
    public UserOptionalListContainer $data;

    /**
     * @var ?string $next
     */
    #[JsonProperty('next')]
    public ?string $next;
}
